packages/preview/colorful-boxes-0.1.0/lib.typ

```diff
@@ -0,0 +1,161 @@
+#let colorbox(title: "title", color: none, radius: 2pt, width: auto, body) = {
+
+  let strokeColor = luma(70)
+  let backgroundColor = white
+
+  if color == "red" {
+    strokeColor = rgb(237, 32, 84)
+    backgroundColor = rgb(253, 228, 224)
+  } else if color == "green" {
+    strokeColor = rgb(102, 174, 62)
+    backgroundColor = rgb(235, 244, 222)
+  } else if color == "blue" {
+    strokeColor = rgb(29, 144, 208)
+    backgroundColor = rgb(232, 246, 253)
+  }
+
+  return box(
+    fill: backgroundColor,
+    stroke: 2pt + strokeColor,
+    radius: radius,
+    width: width
+  )[
+    #block(
+      fill: strokeColor, 
+      inset: 8pt,
+      radius: (top-left: radius, bottom-right: radius),
+    )[
+      #text(fill: white, weight: "bold")[#title]
+    ]
+    #block(
+      width: 100%,
+      inset: (x: 8pt, bottom: 8pt)
+    )[
+      #body
+    ]
+  ]
+}
+
+#let slantedBackground(color: black, body) = {
+  set text(fill: white, weight: "bold")
+  style(styles => {
+    let size = measure(body, styles)
+    let inset = 8pt
+    [#block()[
+      #polygon(
+        fill: color,
+        (0pt, 0pt),
+        (0pt, size.height + (2*inset)),
+        (size.width + (2*inset), size.height + (2*inset)),
+        (size.width + (2*inset) + 6pt, 0cm)
+      )
+      #place(center + top, dy: size.height, dx: -3pt)[#body]
+    ]]
+  })
+}
+
+#let slantedColorbox(title: "title", color: none, radius: 0pt, width: auto, body) = {
```

Comment:
Just wanted to mention that the canonical Typst case would be `slanted-colorbox`, but ultimately it's up to you.